core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -380,7 +380,9 @@ public function show()
 
 				foreach ((array) $value as $v)
 				{
-					$objKey = $this->Database->prepare("SELECT " . Database::quoteIdentifier($chunks[1]) . " AS value FROM " . $chunks[0] . " WHERE id=?")
+					$strRelationField = $objDcaExtractor->getRelations()[$i]['field'] ?? 'id';
```

Comment:
This should be moved outside the `foreach` loop so it isn‘t called unnecessarily.